omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +257,176 @@ else if(identifierSourceType.equals(Identifier_Pool)){
+                List<String> batch = Context.getService(IdentifierSourceService.class).
+                        generateIdentifiers(source, Integer.parseInt(numberToGenerate.toString()), comment.toString());
+                output.add("identifiers", batch.toString());
+                return output;
+            }   
+        }
+        if(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid) instanceof SequentialIdentifierGenerator){
+            SequentialIdentifierGenerator identifierSource = (SequentialIdentifierGenerator) Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
+            Object firstIdentifierBase = updateBody.get("firstIdentifierBase");
+            Object baseCharacterSet = updateBody.get("baseCharacterSet");
+            Object prefix = updateBody.get("prefix");
+            Object suffix = updateBody.get("suffix");
+            Object minLength = updateBody.get("minLength");
+            Object maxLength = updateBody.get("maxLength");
+            Object name = updateBody.get("name");
+            Object description = updateBody.get("description");
+            
+            if (firstIdentifierBase != null && StringUtils.isNotBlank(firstIdentifierBase.toString())) { 
+                identifierSource.setFirstIdentifierBase(firstIdentifierBase.toString());
+            }
+            if (baseCharacterSet != null&& StringUtils.isNotBlank(baseCharacterSet.toString())) { 
+                identifierSource.setBaseCharacterSet(baseCharacterSet.toString());
+            }
+            if (name != null && StringUtils.isNotBlank(name.toString())) {
+                identifierSource.setName(name.toString());
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (prefix != null && StringUtils.isNotBlank(prefix.toString())) { 
+                identifierSource.setPrefix(prefix.toString()); 
+            }
+            if (suffix != null && StringUtils.isNotBlank(suffix.toString())) { 
+                identifierSource.setSuffix(suffix.toString()); 
+            }
+            if (minLength != null && StringUtils.isNotBlank(minLength.toString())) {
+                if(StringUtils.isNumeric(minLength.toString())) {
+                    identifierSource.setMinLength(Integer.parseInt(minLength.toString()));
+                }
+            }
+            if (maxLength != null && StringUtils.isNotBlank(maxLength.toString())) {
+                if(StringUtils.isNumeric(maxLength.toString())) {
+                    identifierSource.setMaxLength(Integer.parseInt(maxLength.toString()));
+                }
+            }
+            updatedIdentifierSource =  save(identifierSource);
+        }
+        if(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid) instanceof IdentifierPool){
```

Comment:
Is it necessary to call this twice? Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid)

Reply:
fixed

Reply:
fixed.